modules/widgets/milestone/milestone.php

```diff
@@ -0,0 +1,360 @@
+<?php
+/*
+Plugin Name: Milestone
+Description: Countdown to a specific date.
+Version: 1.0
+Author: Automattic
+Author URI: http://automattic.com/
+License: GPLv2 or later
+*/
+
+xmpp_message( 'georgestephanis@im.wordpress.com', wp_debug_backtrace_summary() );
+
+class Milestone {
+	public static function init() {
+		add_action( 'widgets_init', array( __class__, 'register_widget' ) );
+	}
+	public static function register_widget() {
+		register_widget( 'Milestone_Widget' );
+	}
+}
+
+Milestone::init();
+
+class Milestone_Widget extends WP_Widget {
+	private static $dir       = null;
+	private static $url       = null;
+	private static $labels    = null;
+	private static $defaults  = null;
+	private static $config_js = null;
+
+	function __construct() {
+		$widget = array(
+			'classname'   => 'milestone-widget',
+			'description' => __( 'Display a countdown to a certain date.' )
+		);
+
+		$control = array(
+			'width' => 251, // Chrome needs a little extra room for the date fields.
+		);
+
+		parent::__construct( 'Milestone_Widget', __( 'Milestone' ), $widget, $control );
+
+		self::$dir = trailingslashit( __DIR__ );
+		self::$url = plugin_dir_url( __FILE__ );
+		self::$labels = array(
+			'year'    => __( 'year' ),
+			'years'   => __( 'years' ),
+			'month'   => __( 'month' ),
+			'months'  => __( 'months' ),
+			'day'     => __( 'day' ),
+			'days'    => __( 'days' ),
+			'hour'    => __( 'hour' ),
+			'hours'   => __( 'hours' ),
+			'minute'  => __( 'minute' ),
+			'minutes' => __( 'minutes' ),
+			'second'  => __( 'second' ),
+			'seconds' => __( 'seconds' ),
+		);
+
+		add_action( 'wp_enqueue_scripts', array( __class__, 'enqueue_template' ) );
+		add_action( 'admin_enqueue_scripts', array( __class__, 'enqueue_admin' ) );
+		add_action( 'wp_footer', array( __class__, 'localize_script' ) );
+
+		if ( is_active_widget( false, false, $this->id_base, true ) || is_active_widget( false, false, 'monster', true ) )
+			add_action( 'wp_head', array( __class__, 'styles_template' ) );
+	}
+
+	public static function enqueue_admin( $hook_suffix ) {
+		if ( 'widgets.php' == $hook_suffix )
+			wp_enqueue_style( 'milestone-admin', self::$url . 'style-admin.css', array(), '20111212' );
+	}
+
+	public static function enqueue_template() {
+		wp_enqueue_script( 'milestone', self::$url . 'milestone.js', array( 'jquery' ), '20160520', true );
+	}
+
+	public static function styles_template() {
+		global $themecolors;
+		$colors = wp_parse_args( $themecolors, array(
+			'bg'     => 'ffffff',
```

Comment:
Abbreviate to 3-character hex codes.  `#ffffff` comes out the same as `#fff` -- no reason to waste the bits.